### PR TITLE
improve installation instructions

### PR DESCRIPTION
**Pull request by gobborg**

### CODE REVIEW

**Comment by gobborg** [comment]
In [commit 547964e](https://github.com/borgbackup/borg-import/pull/52/commits/547964e058c2f372740ed80d36673ac0e4279d8e), I changed all instances of |project_name| to Borg-Import. This was also a problem on [borgbackup/borg](https://github.com/borgbackup/borg/) and it was changed in [this PR](https://github.com/borgbackup/borg/commit/8d830d069f1c168e5402d21f95e44c7614663e3e#diff-8b576d37ea2be85bff0e5e4b366df53a13ffe24bba46b509a280bb0d32005747).

It seems that the global.rst.inc file is not being read anywhere eith either repo, so all automatically generated hyperlinks are broken.

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,6 @@
 MANIFEST
 docs/_build
+borg/
```
Comment (ThomasWaldmann) [inline .gitignore:3]: What's that?

File: docs/installation.rst [hunk 2]
```diff
@@ -5,14 +5,16 @@
 Installation
 ============
 
-To install |project_name|, you need `Python 3`_ and `pip`_ installed. You
-can then either
-`download it directly
+To install Borg-Import, you need `Python 3` and `pip` installed. You
```
Comment (ThomasWaldmann) [inline docs/installation.rst:8]: Yeah, guess these tools are very popular / well known and do not need that linking.

But then also check if there are other links via a global search and also remove the entry in `global.rst.inc`.

File: docs/installation.rst [hunk 3]
```diff
@@ -5,14 +5,16 @@
...
+can then either `download borg-import directly
 <https://github.com/borgbackup/borg-import/archive/master.zip>`_
-or clone it from github_.
```
Comment (ThomasWaldmann) [inline docs/installation.rst:12]: Using "github" as label for that link was a bad idea.

Maybe "our github homepage"?

Also try to use something from global.rst.inc here.

File: docs/installation.rst [hunk 4]
```diff
@@ -5,14 +5,16 @@
...
-Extract the content, open a terminal in its root folder and
-execute the following to install it via pip:
-:code:`pip install --user -e .`
+or clone it from `github<https://github.com/borgbackup/borg-import>`_.
+Unzip the downloaded .zip if necessary.
```
Comment (ThomasWaldmann) [inline docs/installation.rst:12]: .tar.gz is also something that is frequently used, so I'ld prefer keeping the old phrase "extract the content".

File: docs/installation.rst [hunk 5]
```diff
@@ -5,14 +5,16 @@
...
+Open a terminal in the borg-import directory and execute the following to 
+install the program via pip: :code:`pip install --user -e .`
 
 If you have */home/user/.local/bin/* in your ``PATH`` variable,
-you can then start using |project_name|.
+you can then start using Borg-Import.
+Otherwise, you will need to add *.local/bin/* to your ``PATH`` via adding
+``export PATH=$PATH:/home/user/.local/bin/`` to your .bash\_profile file.
```
Comment (ThomasWaldmann) [inline docs/installation.rst:19]: that is bash specific, not everybody uses bash, especially not on the non-linux platforms borg and python also support.

**Review by ThomasWaldmann — CHANGES_REQUESTED** [review]
thanks for your pull request!

it would have been good if this PR only focussed on the replacements done via global.rst.inc (like removing the one for Borg-Import and improving the other ones).

other changes could be also done in a separate PR.

**Comment by ThomasWaldmann** [comment]
Please ignore the py35 test failure, I'll fix that in a separate PR.

**Comment by ThomasWaldmann** [comment]
Please:
- update the master branch in your repo from borg-import master.
- do a `git rebase -i master` in your PR branch afterwards to get rid of the py35 failure and added py310 testing.

File: .gitignore (hunk 1 above)
Comment (gobborg) [reply .gitignore:3]: That is a result of me mistakenly running `pip install -r requirements.d/docs.txt` in the borg-import directory

**Comment by gobborg** [comment]
Can you please confirm the presence or absence of functional github webhooks to autoupdate the [borg-import.readthedocs.io](https://borg-import.readthedocs.io/)?

**Comment by gobborg** [comment]
> thanks for your pull request!
> 
> it would have been good if this PR only focussed on the replacements done via global.rst.inc (like removing the one for Borg-Import and improving the other ones).
> 
> other changes could be also done in a separate PR.

Sorry yeah. I wanted to do 2 PRs, but didn't see a way to do it. My second commit seemed to automatically add to the 1st PR.

File: .gitignore (hunk 1 above)
Comment (ThomasWaldmann) [reply .gitignore:3]: for development, it is usually nice to have a virtual env outside of the project root.

i usually create a virtual env named projectname-env right besides (not inside) the projectname root dir.

File: docs/installation.rst (hunk 2 above)
Comment (gobborg) [reply docs/installation.rst:8]: I'm happy to relink them. I removed them because the links were broken.

File: .gitignore (hunk 1 above)
Comment (gobborg) [reply .gitignore:3]: Updated my reply: I'll delete my local borgimport/borg/ and remove the line from the .gitignore.

**Comment by ThomasWaldmann** [comment]
I added the github integration in rtd and it automatically rebuilt the docs a few minutes ago as a consequence of that.

Let's see if it also rebuilds after we merge this PR.

**Comment by gobborg** [comment]
> I added the github integration in rtd and it automatically rebuilt the docs a few minutes ago as a consequence of that.
> 
> Let's see if it also rebuilds after we merge this PR.

I ask because [the rtd installation page](https://borg-import.readthedocs.io/en/latest/installation.html) doesn't have any of the content that's on the [borg-import branch master installation page](https://github.com/borgbackup/borg-import/blob/master/docs/installation.rst) O_O

**Comment by gobborg** [comment]
> Please:
> 
>     * update the master branch in your repo from upstream borg-import master (== from here).
> 
>     * do a `git rebase -i master` in your PR branch afterwards to get rid of the py35 failure and added py310 testing.
> 
>     * do a `git push -f` to update your PR branch in your github repo

For your first instruction, do you mean rebase gobborg:master from borg-import:master?

File: docs/installation.rst (hunk 3 above)
Comment (gobborg) [reply docs/installation.rst:12]: I changed the global.rst.inc's .git link's "name" and ran `find . -type f | xargs sed -i 's/|git_url|/_repo/g'` to ensure that I didn't break anything in the repo.

File: .gitignore (hunk 1 above)
Comment (ThomasWaldmann) [reply .gitignore:3]: don't see that yet in the PR.

File: docs/global.rst.inc [hunk 6]
```diff
@@ -1,6 +1,5 @@
 .. highlight:: bash
-.. |project_name| replace:: Borg-Import
-.. |git_url| replace:: https://github.com/borgbackup/borg-import.git
+.. _repo replace:: https://github.com/borgbackup/borg-import.git
```
Comment (ThomasWaldmann) [inline docs/global.rst.inc:2]: maybe rather use _githubrepo considering borg also talks about (borg, not git) repos.

File: docs/installation.rst [hunk 7]
```diff
@@ -5,14 +5,13 @@
...
+or clone it from the `repo`_.
+Extract the downloaded .zip if necessary.
+Open a terminal in the borg-import directory and execute the following to install the program via pip:
 :code:`pip install --user -e .`
 
-If you have */home/user/.local/bin/* in your ``PATH`` variable,
-you can then start using |project_name|.
+If you have */home/user/.local/bin/* in your ``PATH`` variable, you can then start using Borg-Import.
+Otherwise, you will need to add *.local/bin/* to your ``PATH`` via adding ``export PATH=$PATH:/home/user/.local/bin/``
+to your .profile file if you are using a POSIX-based shell. Don't forget to :code:`source ~/.profile` when you're done.
```
Comment (ThomasWaldmann) [inline docs/installation.rst:17]: https://linuxtect.com/zsh-profile-file-profile-zprofile-zshrc/ seems like the instructions would not work for macOS.

Can we just keep 17/18 as is and assume all users can find out elsewhere how to set PATH for their OS?

**Comment by ThomasWaldmann** [comment]
> I ask because [the rtd installation page](https://borg-import.readthedocs.io/en/latest/installation.html) doesn't have any of the content that's on the [borg-import branch master installation page](https://github.com/borgbackup/borg-import/blob/master/docs/installation.rst) O_O

Looks ok for me!?

**Comment by gobborg** [comment]
> > I ask because [the rtd installation page](https://borg-import.readthedocs.io/en/latest/installation.html) doesn't have any of the content that's on the [borg-import branch master installation page](https://github.com/borgbackup/borg-import/blob/master/docs/installation.rst) O_O
> 
> Looks ok for me!?

Looks ok for me now, too... no idea why!

**Review by ThomasWaldmann — APPROVED** [review]
lgtm

**Comment by gobborg** [comment]
Glad to hear it. Excited to find out if the rtd updates appropriately.

**Comment by ThomasWaldmann** [comment]
thanks!

**Comment by gobborg** [comment]
Looks good! I'm going to continue improving the borg-import docs. How do you want me to submit PRs?

**Comment by ThomasWaldmann** [comment]
Focussed (and in case of "code" changes: tested) PRs are very welcome!

**Pull request merged**